src/Controller/AgendaBroadcastController.php

```diff
@@ -96,26 +95,20 @@ public function broadcastAgenda(Agenda $agenda, CprHelper $cprHelper, DigitalPos
             // Move file
             $updatedFileName = $documentUploader->uploadFile($fileName);
 
-            // Create document
-            $document = new Document();
-            $document->setFilename($updatedFileName);
-            $document->setDocumentName($agendaBroadcast->getTitle());
-            $document->setPath($documentUploader->getFilepathFromProjectDirectory($updatedFileName));
-
             /** @var User $user */
             $user = $this->getUser();
-            $document->setUploadedBy($user);
-            $document->setType(new TranslatableMessage('Agenda broadcast', [], 'agenda'));
 
-            $entityManager->persist($document);
+            // Create document
+            $newDocument = $documentUploader->createDocumentFromPath($updatedFileName, $user, $agendaBroadcast->getTitle(), 'Agenda broadcast');
```

Comment:
Why is it called “newDocument”? Previously, we didn't use `$newDocument = new Document()` even though that document was also new.

Reply:
It was initially done in the `DocumentController` as this already had a `$document` variable. 

I will change it back in controllers where this is not the case.